eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/batch/io/ActiveAccessionsVariantReader.java

```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+
+    @Override
+    public List<Variant> read() {
+        List<Long> cveAccList = new ArrayList<>();
+        String line;
+
+        try {
+            while (cveAccList.size() < chunkSize && (line = reader.readLine()) != null) {
+                String rsAcc = line.split("[ \t]+")[0].trim();
+                if (!rsAcc.isEmpty()) {
+                    cveAccList.add(Long.parseLong(rsAcc));
+                }
+            }
+            if (cveAccList.isEmpty()) {
+                return null;
+            } else {
+                return processCveAccession(cveAccList);
+            }
+        } catch (IOException e) {
+            throw new ItemStreamException("Error reading variant Accessions from file", e);
+        }
+    }
+
+
+    public List<Variant> processCveAccession(List<Long> cveAccList) {
+        Set<Long> cveAccSet = new HashSet<>(cveAccList);
+
+        List<ClusteredVariantEntity> clusteredVariantEntityList = getClusteredVariantEntities(cveAccSet);
+        List<SubmittedVariantEntity> submittedVariantEntityList = getSubmittedVariantEntities(cveAccSet);
+
+        Map<Long, List<SubmittedVariantEntity>> submittedVariantEntityMap = submittedVariantEntityList.stream()
+                .collect(Collectors.groupingBy(SubmittedVariantEntity::getClusteredVariantAccession));
+
+        List<Variant> variantList = new ArrayList<>();
+
+        for (ClusteredVariantEntity clusteredVariant : clusteredVariantEntityList) {
+            List<SubmittedVariantEntity> submittedVariantEntities = submittedVariantEntityMap.get(clusteredVariant.getAccession());
+            if (submittedVariantEntities != null) {
+                variantList.addAll(getVariants(clusteredVariant, submittedVariantEntities));
+            }
+        }
+
+        return variantList;
+    }
+
+    private List<ClusteredVariantEntity> getClusteredVariantEntities(Set<Long> cveAccs) {
+        Query query = query(where(CVE_ACC_FIELD).in(cveAccs).and(CVE_ASSEMBLY_FIELD).is(assembly));
+        List<ClusteredVariantEntity> evaResults = mongoTemplate.find(query, ClusteredVariantEntity.class);
+        List<DbsnpClusteredVariantEntity> dbsnpResults = mongoTemplate.find(query, DbsnpClusteredVariantEntity.class);
+        return Stream.concat(evaResults.stream(), dbsnpResults.stream()).collect(Collectors.toList());
+    }
+
+    private List<SubmittedVariantEntity> getSubmittedVariantEntities(Set<Long> cveAccs) {
+        Query query = query(where(SVE_RS_FIELD).in(cveAccs).and(SVE_ASSEMBLY_FIELD).is(assembly).and(SVE_TAX_FIELD).is(taxonomy));
+        List<SubmittedVariantEntity> evaResults = mongoTemplate.find(query, SubmittedVariantEntity.class);
+        List<DbsnpSubmittedVariantEntity> dbsnpResults = mongoTemplate.find(query, DbsnpSubmittedVariantEntity.class);
+        return Stream.concat(evaResults.stream(), dbsnpResults.stream()).collect(Collectors.toList());
```

Comment:
It might never be a problem but this function can in theory return a very large number of variant creating a memory issue. It will be especially true in the merged and deprecated use case, less so in the case of active variants.